File: Data Structures & Algorithms/two-integer-sum/submission-6.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numHash = {}

        for i in range(len(nums)):
            numHash[nums[i]] = i
        
        for i in range(len(nums)):
            numCompliment = target - nums[i]

            if numCompliment in numHash:
                if i == numHash[numCompliment]:
                    continue
                return [i, numHash[numCompliment]]
        
        return []

        

File: Data Structures & Algorithms/decode-ways/submission-6.cpp
class Solution {
public:

    int numDecodings(string s) {
        if (s.size() == 0) {
            return 1;
        }

        return dp(0, s);
    }

    int dp(int i, string& s) {
        if (i >= (s.size())) {
            return 1;
        }

        if (s[i] == '1') {
            if (i < (s.size() - 1)) {
                return dp(i+1, s) + dp(i+2, s);
            }
        }

        if (s[i] == '2') {
            if (i < (s.size() - 1) && s[i+1] <= '6') {
                return dp(i+1, s) + dp(i+2, s);
            }
        }

        if (s[i] != '0') {
            return dp(i+1, s);
        }

        return 0;
    }
};
